src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -291,8 +297,147 @@ OCC::Optional<OCC::PinStateEnums::PinState> OCC::CfApiWrapper::PlaceHolderInfo::
     return cfPinStateToPinState(_data->PinState);
 }
 
-OCC::Result<void, QString> OCC::CfApiWrapper::registerSyncRoot(const QString &path, const QString &providerName, const QString &providerVersion)
+QString convertSidToStringSid(void *sid)
+{
+    QString result;
+
+    wchar_t *stringSid = nullptr;
+    if (::ConvertSidToStringSid(sid, &stringSid)) {
+        result = QString::fromWCharArray(stringSid);
+    }
+
+    if (stringSid) {
+        ::LocalFree(stringSid);
+        stringSid = nullptr;
+    }
+
+    return result;
+}
+
+QString retrieveWindowsSID()
+{
+    // FYI: code here is mostly identical to Windows-Classic-Examples/Samples/CloudMirror
+    QScopedPointer<TOKEN_USER> tokenInfo;
```

Comment:
I'd favor std::unique_ptr these days. Less error prone.

Reply:
@er-vin Ok. Bringing that back. I thought it makes no sense to create a method that returns a pointer of that token like it was done originally in that MS's sample. But, ok, let's bring back the old `unique_ptr` returning function then.